.github/workflows/tests.yml

```diff
@@ -0,0 +1,93 @@
+on:
+  - pull_request
+  - push
+
+name: Tests
+
+jobs:
+  unit:
+    name: Unit and Integration tests
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          # - windows-latest
+
+        php-version:
+          - "8.0"
+
+        dependencies:
+          - lowest
+          - highest
```

Comment:
Doesn't `locked` mean using `composer.lock`? If so, I think libraries shouldn't publish the lock file.

Reply:
Yes, `locked` means using a committed `composer.lock`.

You can put your `composer.lock` in `export-ignore`, and it will be ignored regardless by downstream `composer install` and `composer update` operations.

The reasons why `composer.lock` is valuable is that tools like `vimeo/psalm`, `phpunit/phpunit` and `phpcs/php_codesniffer` often contain bugfixes and tiny improvements that completely destabilize the whole downstream consumer CI pipelines. This is also not fixable on their end: if `vimeo/psalm` slightly improves the type inference of an internal PHP function, for example, you might get reported `RedundantCondition` or similar errors in your pipeline.

By running them **locked**, you can come back to this project one year from now:

 * `latest` pipeline will have "discovered" any changes that might be problematic
 * `lowest` pipeline might highlight how some dependencies no longer work with your code changes (signatures, mostly)
 * `locked` will allow you and any contributor to just clone the repo, run `composer install` and start working without worrying too much about unrelated failures.

I therefore encourage to run these specific tasks only with `locked`:

 * coverage
 * static analysis
 * mutation tests
 * code style checks

For phpunit itself, you can run `latest` + `lowest` + `locked`.

Important for this to work is to set up @dependabot for your project. See for example:

 * https://github.com/Roave/BackwardCompatibilityCheck/pull/320/files#diff-dd4fbda47e51f1e35defb9275a9cd9c212ecde0b870cba89ddaaae65c5f3cd28
 * https://github.com/Roave/BackwardCompatibilityCheck/pull/320/files#diff-82f8802c05df24bb863565fa5512918da7cdced6a3395798ed3afbe58ca9c230

Reply:
See also https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2021-08-02-TSC-Minutes.md#locked-dependencies-for-all-supported-php-versions